### ISSUE TITLE
Add support for links `<a>` with `download` attribute

### ISSUE THREAD

**Issue by ShGKme** [electron/electron#44375]
### Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a feature request that matches the one I want to file, without success.

### Problem Description

Click on a `<a download>` link:
- Does not trigger download and corresponding `will-download`, but navigates to the link
- Triggers `will-navigate`, but doesn't pass any additional information via event to know it was a download link for manual handling

As the result, download links works like a general links, and there is no way to handle it from the Electron side.

Both limitations require modifying a web-page to handle download link. For example, via
```js
// Must be added to the web page
document.addEventListener('click', (event) => {
	const link = event.target.closest('a')
	if (link && link.hasAttribute('download')) {
		event.preventDefault()
		window.IPC.downloadURL(link.href)
	}
})
```

However, this won't work with a popular download trigger method from JS and used by many libraries:
```js
const hiddenElement = document.createElement('a')
hiddenElement.download = ''
hiddenElement.href = url
hiddenElement.click() // Doesn't trigger document click
```

In other words, it is not possible to handle `download` on an app-level in a general case.

### Proposed Solution

Click on a `<a download>` link should trigger download that can be handled via `will-download` event.

### Alternatives Considered

Event in `will-navigate` should include an information, that is was a download link. For example, via a new `disposition` value.

### Additional Information

There is a closed related issue (https://github.com/electron/electron/issues/1909). It was closed by adding the download manager API. However, this API doesn't add actual support for download links.

**Comment by nikwen** [comment]
If this is implemented, should there be an equivalent to `webContents.setWindowOpenHandler(handler)` to be able to deny all downloads?

Seems like allowing arbitrary downloads could be relevant for security.

**Comment by TheOneTheOnlyJJ** [comment]
> If this is implemented, should there be an equivalent to `webContents.setWindowOpenHandler(handler)` to be able to deny all downloads?

There should be a way to control what and when downloads can happen.

**Comment by ShGKme** [comment]
> If this is implemented, should there be an equivalent to `webContents.setWindowOpenHandler(handler)` to be able to deny all downloads?

It could trigger `will-download` to handle downloading manually.

**Comment by ShGKme** [comment]
Additional note: `download` attribute should be used for filename suggestion in the save dialog like in web-browser.

Currently it is non-trivial with `will-download`. You need to provide it from IPC handler to `will-download` handler via some intermediate thing.

**Comment by jikrana-del** [comment]
Skip custom click handling for links with download attribute
